change_settings/change_settings_functionality.php

```diff
@@ -0,0 +1,96 @@
+<?php
+session_start();
+include "../shared/php/database.php";
+
+header("Content-Type: application/json");
+$request = file_get_contents("php://input");
+$conn = connect_local();
+
+if (isset($request) && !empty($request)) {
```

Comment:
Your else case is missing:
else {
    echo json_encode(["result" => null, "error" => true, "errorText" => "Request is not set or is empty."]);
}